File: Source/Fantasy.Jobs/TrackMessageService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Fantasy.ServiceModel;
using Fantasy.Tracking;

namespace Fantasy.Jobs
{
    public class TrackMessageService : AbstractMessageService, IService, IObjectWithSite 
    {
        public override System.Windows.Forms.DialogResult Show(System.Windows.Forms.DialogResult defaultResult, System.Windows.Forms.IWin32Window owner, string text, string caption, System.Windows.Forms.MessageBoxButtons buttons, System.Windows.Forms.MessageBoxIcon icon, System.Windows.Forms.MessageBoxDefaultButton defaultButton, System.Windows.Forms.MessageBoxOptions options)
        {
            if (_track != null)
            {
                _track["message"] = text;

            }

            return defaultResult; 
        }

        public override void WriteLine(string text)
        {
            _track["message"] = text;
        }

        #region IService Members

        private ITrackProvider _track;

        public void InitializeService()
        {
            _track = this.Site.GetService<ITrackProvider>();
            if (Initialize != null)
            {
                Initialize(this, EventArgs.Empty);
            }
        }

        public void UninitializeService()
        {
            if (Uninitialize != null)
            {
                Uninitialize(this, EventArgs.Empty);
            }
        }

        public event EventHandler Initialize;

        public event EventHandler Uninitialize;

        #endregion

        #region IObjectWithSite Members

        public IServiceProvider Site
        {
            get;
            set;
        }

        #endregion
    }
}

File: branches/Stheno/Fantasy.Jobs/src/fantasy/jobs/TrackMessageService.java
package fantasy.jobs;

import fantasy.servicemodel.*;
import Fantasy.Tracking.*;

public class TrackMessageService extends AbstractMessageService implements IService, IObjectWithSite
{
	@Override
	public int Show(int defaultResult, System.Windows.Forms.IWin32Window owner, String text, String caption, System.Windows.Forms.MessageBoxButtons buttons, System.Windows.Forms.MessageBoxIcon icon, System.Windows.Forms.MessageBoxDefaultButton defaultButton, System.Windows.Forms.MessageBoxOptions options)
	{
		if (_track != null)
		{
			_track["message"] = text;

		}

		return defaultResult;
	}

	@Override
	public void WriteLine(String text)
	{
		_track["message"] = text;
	}

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#region IService Members

	private ITrackProvider _track;

	public final void InitializeService()
	{
		_track = this.getSite().<ITrackProvider>GetService();
		if (Initialize != null)
		{
			Initialize(this, EventArgs.Empty);
		}
	}

	public final void UninitializeService()
	{
		if (Uninitialize != null)
		{
			Uninitialize(this, EventArgs.Empty);
		}
	}

//C# TO JAVA CONVERTER TODO TASK: Events are not available in Java:
//	public event EventHandler Initialize;

//C# TO JAVA CONVERTER TODO TASK: Events are not available in Java:
//	public event EventHandler Uninitialize;

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#endregion

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#region IObjectWithSite Members

	private IServiceProvider privateSite;
	public final IServiceProvider getSite()
	{
		return privateSite;
	}
	public final void setSite(IServiceProvider value)
	{
		privateSite = value;
	}

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#endregion
}